x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/query/LongScriptFieldDistanceFeatureQuery.java

```diff
@@ -0,0 +1,211 @@
+                @Override
+                public boolean matches() throws IOException {
+                    script.runForDoc(approximation().docID());
+                    return script.count() > 0;
+                }
+
+                @Override
+                public float matchCost() {
+                    return MATCH_COST;
+                }
+            };
+            disi = TwoPhaseIterator.asDocIdSetIterator(twoPhase);
+            this.weight = LongScriptFieldDistanceFeatureQuery.this.boost * boost;
+        }
+
+        @Override
+        public int docID() {
+            return disi.docID();
+        }
+
+        @Override
+        public DocIdSetIterator iterator() {
+            return disi;
+        }
+
+        @Override
+        public TwoPhaseIterator twoPhaseIterator() {
+            return twoPhase;
+        }
+
+        @Override
+        public float getMaxScore(int upTo) throws IOException {
+            return boost;
+        }
+
+        @Override
+        public float score() throws IOException {
+            if (script.count() == 0) {
+                return 0;
+            }
+            return LongScriptFieldDistanceFeatureQuery.this.score(weight, (double) minAbsoluteDistance(script));
+        }
+    }
+
+    long minAbsoluteDistance(AbstractLongScriptFieldScript script) {
+        long minDistance = Long.MAX_VALUE;
+        for (int i = 0; i < script.count(); i++) {
+            minDistance = Math.min(minDistance, distanceFor(script.values()[i]));
+        }
+        return minDistance;
+    }
+
+    long valueWithMinAbsoluteDistance(AbstractLongScriptFieldScript script) {
+        long minDistance = Long.MAX_VALUE;
+        long minDistanceValue = Long.MAX_VALUE;
+        for (int i = 0; i < script.count(); i++) {
+            long distance = distanceFor(script.values()[i]);
+            if (distance < minDistance) {
+                minDistance = distance;
+                minDistanceValue = script.values()[i];
+            }
+        }
+        return minDistanceValue;
+    }
+
+    long distanceFor(long value) {
+        long distance = Math.max(value, origin) - Math.min(value, origin);
+        if (distance < 0) {
+            // The distance doesn't fit into signed long so clamp it to MAX_VALUE
+            return Long.MAX_VALUE;
+        }
+        return distance;
+    }
+
+    float score(float weight, double distance) {
+        return (float) (weight * (pivot / (pivot + distance)));
+    }
+
+    @Override
+    public String toString(String field) {
+        StringBuilder b = new StringBuilder();
+        if (false == fieldName().equals(field)) {
+            b.append(fieldName()).append(":");
+        }
+        b.append(getClass().getSimpleName()).append("(origin=").append(origin).append(",pivot=").append(pivot).append(")");
+        return b.toString();
+
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), origin, pivot);
```

Comment:
should we incorporate the initial `boost` to `hashCode`, `equals`, `toString` ?

Reply:
Yes.